object/state.go

```diff
@@ -175,17 +175,43 @@ func (e *Environment) SaveGlobals(to io.Writer, maxValueLen int) (int, error) {
 	return n, nil
 }
 
+func (e *Environment) makeRef(name string) (*Reference, bool) {
+	orig := e
+	for e.outer != nil {
+		obj, ok := e.outer.store[name]
+		if ok {
+			for obj.Type() == REFERENCE { // for is in theory not needed as we deref new refs.
+				obj = obj.(Reference).Value()
+			}
+			ref := Reference{Name: name, RefEnv: e.outer}
+			orig.store[name] = ref
+			orig.getMiss++ // creating a ref is a miss.
+			return &ref, true
+		}
+		e = e.outer
```

Comment:
```suggestion
		obj, ok := e.outer.store[name]
		if !ok {
			e = e.outer
			continue
		}
		for obj.Type() == REFERENCE { // for is in theory not needed as we deref new refs.
			obj = obj.(Reference).Value()
		}
		ref := Reference{Name: name, RefEnv: e.outer}
		orig.store[name] = ref
		orig.getMiss++ // creating a ref is a miss.
		return &ref, true
		
```

Reply:
hmm I just explained why the code shouldn't work :)